app/localize.js

```diff
@@ -0,0 +1,27 @@
+var languages = {
+    "en_US": {
+        "name": "English", "messages": require.main.require("./locales/en_US.json")
+    }, "fr_FR": {
+        "name": "Français", "messages": require.main.require("./locales/fr_FR.json")
+    }
+};
+
+var configKey = "window:language";
+
+var localize = function (config, key) {
```

Comment:
Have you considered to directly give the expected language code as a parameter (instead of the full `config` object)? I think things would be more independent this way.

```javascript
function localize (lang, key) {
	// 1. Check if app/locales/[lang].json exists and load it if it does (otherwise return null)
    // 2. Check if key has a translation and return it (else return null)
}
```
In order to avoid loading JSON on each call, we can curry the function in order to call it once and for all in `abr-menu.js`:

```javascript
function localize (lang) {
	// Load app/locales/[lang].json and return:
	return function (key) {
    	// Check if key has a translation and return it (else return null)
    }    
}
```

In this solution, `availableLanguages` is no longer needed since languages list can be get by simply looking at language JSON files in `locales` dir.

Of course this one possible solution to make things more modular and reusable, but maybe you will think about a better way to perform this.